src/client.js

```diff
@@ -93,7 +93,53 @@ export function bootstrap() {
     )
     : new URL(link.getAttribute("href"));
   const client = new Client(initialURL.href);
-  // Register a global listener for history updates.
+
+  // Add click event listener to div#app
+  const appDiv = document.getElementById("app");
+  console.assert(!!appDiv, "bootstrap error: missing div#app element");
+  appDiv.addEventListener("click", (event) => {
+    // Check if the target is an anchor element
+    if (!(event.target instanceof HTMLAnchorElement)) return;
+
+    const anchor = event.target;
+    const href = anchor.href; // Safe to access href directly
+
+    // Check if href is external or has a non-web protocol
+    let url;
+    try {
+      url = new URL(href, appDiv.baseURI);
+    } catch {
+      console.warn(`Invalid href: ${href}`);
+      return;
+    }
+    const isExternal = url.protocol !== "http:" && url.protocol !== "https:" ||
+      url.origin !== new URL(appDiv.baseURI).origin;
```

Comment:
Not critical to change now but this could just be `isKnownProtocol` since we also do the isExternal check by comparing the origins inside `follow`